src/app/features/room/RoomInput.tsx

```diff
@@ -227,17 +227,18 @@ export const RoomInput = forwardRef<HTMLDivElement, RoomInputProps>(
       const contentsPromises = uploads.map(async (upload) => {
         const fileItem = selectedFiles.find((f) => f.file === upload.file);
         if (!fileItem) throw new Error('Broken upload');
+        const caption = toPlainText(editor.children).trim() || undefined;
```

Comment:
It looks like following changes add same caption to multiple uploaded file. Do we really want that? Can't we just send it as separate in that case?

Reply:
Would it not be better to attach the caption to the last uploaded item, rather than as an entirely separate message?

Reply:
<img width="835" alt="image" src="https://github.com/user-attachments/assets/2f18e101-dddb-4f7e-883a-67b91fb1e037">
I think captions should be added here for each file and message input should not be used for caption as doing so will break the current functionality. For example someone wants to send an image right after writing their TWIM message.

Reply:
Yeah adding full message composer won't be a good idea, but maybe we can add plaintext caption as file rename option, not sure about how that would go. So if we don't want to go with sending caption from main composer yet could we please limit this PR for display only support as that alone would be really helpful.

Reply:
I have now removed sending captions altogether. Rendering plaintext & HTML captions remains.

For the sake of satisfying other clients, `body` is still the filename (since it's required) but the `filename` field is still sent with the same value.